gateway-service-impl/src/main/java/org/hypertrace/gateway/service/entity/query/SelectionAndFilterNode.java

```diff
@@ -0,0 +1,53 @@
+package org.hypertrace.gateway.service.entity.query;
+
+import java.util.List;
+import org.hypertrace.gateway.service.entity.query.visitor.Visitor;
+import org.hypertrace.gateway.service.v1.common.OrderByExpression;
+
+public class SelectionAndFilterNode implements QueryNode {
+
+  private final String source;
+  private final int limit;
+  private final int offset;
+  private final List<OrderByExpression> orderBys;
+
+  public SelectionAndFilterNode(String source, int limit, int offset, List<OrderByExpression> orderBys) {
+    this.source = source;
+    this.limit = limit;
+    this.offset = offset;
+    this.orderBys = orderBys;
+  }
+
+  @Override
+  public <R> R acceptVisitor(Visitor<R> v) {
+    return v.visit(this);
```

Comment:
All the nodes' implementation of this method seems to be exactly same :) Why? Is the inheritance usage and abstractions correctly defined? Why am I not seeing the special execution logic related to this node in this class?

Reply:
The Visitor interface works on the concrete node implementations. So we need a handle to `this` object for the visit method. I dont think there is a better way to do this.

```
public interface Visitor<R> {
  R visit(DataFetcherNode dataFetcherNode);

  R visit(AndNode andNode);

  R visit(OrNode orNode);

  R visit(SelectionNode selectionNode);

  R visit(SortAndPaginateNode sortAndPaginateNode);

  R visit(NoOpNode noOpNode);
}
```